src/MainWindow.vala

```diff
@@ -0,0 +1,78 @@
+public class KillSwitch.MainWindow : Hdy.ApplicationWindow {
+    public MainWindow (Gtk.Application application) {
+        Object(
+            application: application,
+            resizable: false,
+            width_request: 480,
+            height_request: 320
+        );
+    }
+
+    construct {
+        Hdy.init ();
+
+        var headerbar = new Hdy.HeaderBar () {
+            show_close_button = true,
+            decoration_layout ="close:"
+        };
+        headerbar.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+        headerbar.get_style_context ().add_class ("default-decoration");
+
```

Comment:
The `Granite.Widgets.Welcome` widget was not used here because it uses a slightly darker background color which is not the same as the flat headerbar color. So it doesn't look like as if the headerbar is part of the window itself.